### PR TITLE
fixes some leaderboard formatting issues mostly

### PR DESCRIPTION
**Pull request by sunglasseds**
was having this issue in a server im in (friend had a "`" in their name and the fishy leaderboard was getting all fucked up as a result) and i didnt want to make it an issue because its a really small bug but if youd rather fix this your own way i dont really care, as long as this gets resolved somehow. not every instance that this could happen is covered mostly because i didnt want to spend my day scanning the codebase for formatting strings and thinking of ways that things could go wrong

### CODE REVIEW

File: helpers/utilityfunctions.py [hunk 1]
```diff
@@ -654,3 +654,9 @@ def previous(self):
 
     def current(self):
         return self.items[self.index]
+
+# people with `, *, etc, in their usernames break a ton of formatting
+def escape_formatting_chars(s):
+    for c in ["*", "_", "`", "~", ">"]: # frankly im not sure if > has any effect in embeds but i'll leave it here
```
Comment (Saragon4005) [inline helpers/utilityfunctions.py:660]: Technically surrounding a link with < and > will stop it from expanding, but this only applies to things discord detects as links, so aside from very rear cases like a user having <www.example.com> in their usernames it's not going to be absorbed

**Comment by joinemm** [comment]
Looks good but, there already is `escape_md()` function in `utilityfunctions.py` which does exactly this using regex. Just needs to be applied to more places. Currently I think it's only used for lastfm artist/album/track names.

Also this code wouldn't work since the function is undefined without specifying `util.escape_formatting_chars()`

**Comment by joinemm** [comment]
@sunglasseds If you want to make those changes I can then merge it.

**Comment by joinemm** [comment]
closing due to inactivity, functionality added in 28d2e5e

**Pull request closed without merging**